### PR TITLE
Uniprot sequence space fix

### PR DESCRIPTION
**Pull request by emckee2006**

### CODE REVIEW

**Comment by josemduarte** [comment]
Could you give a brief explanation on what this PR is about? It doesn't look like a space fix only

**Comment by emckee2006** [comment]
The 2nd commit is different and could go in a different PR. All im doing in
the 2nd commit is refactoring the ugly code I'd written last PR that was
merged into something more readable/maintainable...

File: biojava-core/src/main/java/org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.java [hunk 1]
```diff
@@ -151,13 +151,13 @@ public void setContents(String sequence) throws CompoundNotFoundException {
 		// Horrendously inefficient - pretty much the way the old BJ did things.
 		// TODO Should be optimised.
 		// NOTE This chokes on whitespace in the sequence, so whitespace is stripped
-		this.sequence = sequence.replaceAll("\\s","");
+		this.sequence = sequence.replaceAll("\\s", "").trim();
```
Comment (josemduarte) [inline biojava-core/src/main/java/org/biojava/nbio/core/sequence/loader/UniprotProxySequenceReader.java]: trim() is redundant here, the leading/trailing spaces are already replaced.

**Comment by josemduarte** [comment]
@heuermh ok to merge?

**Pull request merged**